frontend/src/Teacher/InbusImport.vue

```diff
@@ -0,0 +1,322 @@
+  const [dept_code, version] = svcc.split('/');
+  const [dept, code] = dept_code.split('-');
+
+  const svcc_str = dept + code + version;
+  const svcc_num = Number(svcc_str);
+
+  return svcc_num;
+}
+
+function sortCollection<T>(collection: T[], key: string) {
+  collection.sort((a, b) => {
+    const name_a = a[key].toUpperCase();
+    const name_b = b[key].toUpperCase();
+    if (name_a < name_b) {
+      return -1;
+    }
+    if (name_a > name_b) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
+async function loadKelvinSubjects(): Promise<KelvinSubject[]> {
+  const res = await fetch('/api/subjects/all', {});
+  const subjects_kelvin_resp = await res.json();
+
+  const subjects_kelvin: KelvinSubject[] = subjects_kelvin_resp.subjects;
+  sortCollection(subjects_kelvin, 'name');
+
+  return subjects_kelvin;
+}
+
+async function loadInbusSubjects(kelvin_subjects: KelvinSubject[]): Promise<InbusSubjectVersion[]> {
+  const res = await fetch('/api/inbus/subject_versions', {});
+
+  const subjects_inbus: InbusSubjectVersion[] = await res.json();
+  const subject_kelvin_abbrs = kelvin_subjects.map((s) => s.abbr);
+
+  const subjects_inbus_filtered = subjects_inbus.filter((subject_inbus) =>
+    subject_kelvin_abbrs.includes(subject_inbus.subject.abbrev)
+  );
+  subjects_inbus_filtered.sort(
+    (a, b) => svcc2num(a.subjectVersionCompleteCode) - svcc2num(b.subjectVersionCompleteCode)
+  );
+
+  return subjects_inbus_filtered;
+}
+
+function parseSemesters(semesters_data: Semester[]) {
+  const semesters = semesters_data.map((sm) => ({
+    pk: sm.pk,
+    year: sm.year,
+    winter: sm.winter,
+    display: String(sm.year) + (sm.winter ? 'W' : 'S')
+  }));
+
+  return semesters;
+}
+
+async function loadSemesters() {
+  const request = assembleRequest('/api/semesters');
+  const res = await fetch(request, {});
+
+  const semesters_data = await res.json();
+  return parseSemesters(semesters_data['semesters']);
+}
+
+async function loadScheduleForSubjectVersionId(subject_index: number) {
+  subject_inbus_schedule.value = null;
+  activities_to_teacher_selected.value = {};
+
+  const versionId = subjects_inbus_filtered[subject_index].subjectVersionId;
+  const request = assembleRequest(`/api/inbus/schedule/subject/version/${versionId}`);
+  const res = await fetch(request, {});
+  subject_inbus_schedule.value = await res.json();
+}
+
+async function loadTeachers(): Promise<Teacher[]> {
+  const request = assembleRequest('/api/teachers/all');
+  const res = await fetch(request, {});
+  const teachers_data = await res.json();
+  const teachers: Teacher[] = teachers_data['teachers'];
+
+  sortCollection(teachers, 'last_name');
+
+  return teachers;
+}
+
+async function importActivities() {
+  busy.value = true;
+  const req = {
+    semester_id: semester.value,
+    subject_abbr: subject_kelvin_selected.value,
+    activities: classes_to_import.value,
+    activities_to_teacher: activities_to_teacher_selected.value
```

Comment:
Note that `activities_to_teacher_selected` is empty by default, it will **NOT** be pre-filled with the first teacher which is then shown in the UI. That means that unless you explicitly select a teacher for the lesson, it will be missing and the bulk import will fail.

Therefore, the backend should either use fallible indexing for it, and still set some default teacher, or the UI should not allow importing the subject until you explicitly select a teacher (or alternatively, the UI should just pre-fill the first teacher from the list automatically).

Reply:
OK, hope it's fixed in the current version.